docs/proposals/updates-first-pass.md

```diff
@@ -0,0 +1,205 @@
+spec, the `next_values` are not wrapped in that key specifically). It allows
+for the APBs to define "this is how we will move from our current state, to the
+next". Example request body taken from upstream spec, let's refer to this as
+the `UpdatePayload`:
+
+```json
+{
+  "context": {
+    "platform": "cloudfoundry",
+    "some_field": "some-contextual-data"
+  },
+  "service_id": "service-guid-here",
+  "plan_id": "plan-guid-here",
+  "parameters": {
+    "parameter1": 1,
+    "parameter2": "foo"
+  },
+  "previous_values": {
+    "plan_id": "old-plan-guid-here",
+    "service_id": "service-guid-here",
+    "organization_id": "org-guid-here",
+    "space_id": "space-guid-here"
+  }
+}
+```
+
+After validating a given request, the broker will pass this payload through to
+an APB, triggering a new "update" action that should execute a new `update.yml`
+playbook. See APB section for more.
+
+#### New Credentials
+
+It's possible for param updates or plan transitions to mutate credentials
+as a result of an update. Ex: database plan transition needs to issue new
+database credentials as a result of the update
+
+NOTE: Since the first-pass is limited to updates of instances without outstanding
+bindings, the only possible active credentials in the broker are those from
+a provision.
+
+Therefore, the broker *must* monitor an update APB with existing `ext_creds`
+functions, looking for newly issued creds.
+
+If credentials are discovered, the newly issued object should be merged with
+the existing credentials, and then reinserted into etcd as the new, official
+credentials object. Ex:
+
+**original creds**
+```json
+{
+  "db_name": "foobar_db",
+  "db_user": "duder",
+  "db_pass": "topsecret"
+}
+```
+
+**updated creds; extracted from update APB run**
+```json
+{
+  "db_user": "duder_prime",
+  "db_pass": "supertopsecret"
+}
+```
+
+**new creds; merge(updated, original)**
+```json
+{
+  "db_name": "foobar_db",
+  "db_user": "duder_prime",
+  "db_pass": "supertopsecret"
+}
+```
+
+* Q: Is there any value in versioning this object? Ex: `_apb_creds_version` field
+somewhere that begins at 1 and increments each time an update occurs?
+
+#### APB
+
+Overall, the goal with update should be to push the complexity of what an update
+actually means for an APB into the APB ansible itself. APBs supporting update
+must now define a new playbook similar to the other actions called `update.yml`. This
+playbook should define the APB's update behavior given the `UpdatePayload` from
+from the broker.
+
+**plan transitions**
+
+The OSB spec explicitly states the broker must reject plan transitions that are
+invalid. Effectively, this is a directed graph with transitions as edges and
+plans as the nodes. APBs will need to define this graph in their `apb.yml`
+so the broker is able to validate a requested transition.
+
+Proposal is to add an optional `updates_to` list on `Plan` objects that indicates
```

Comment:
ACK, like the flexibility `update_from` supports. It feels a little backwards in my mind but your arguments are solid.

WRT versions, I think we need to think through and possibly decouple versions from plans [for these reasons](https://github.com/openshift/ansible-service-broker/pull/368#issuecomment-323755251). At the very least, that sounds like a problem discussion in-scope for the next pass proposal.